pkg/apis/kubebind/v1alpha1/apiserviceexport_types.go

```diff
@@ -90,8 +90,25 @@ type APIServiceExportSpec struct {
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="informerScope is immutable"
 	InformerScope Scope `json:"informerScope"`
+
+	// ClusterScopedIsolation specifies how cluster scoped service objects are isolated between multiple consumers on the provider side.
+	// It can be "Prefixed", "Namespaced", or "None".
+	ClusterScopedIsolation Isolation `json:"clusterScopedIsolation"`
```

Comment:
is this required or can we define a default? Depends a lot on the backend. So maybe a default is not good, but rather force the backend to set this explicitly. So maybe required would be good if `apiextensionsv1.ResourceScope` below is set to `Cluster`. CEL can do this.

This also depends on whether the

Reply:
Thanks! I realized that the `ClusterScopedIsolation` field is required _only_ when the resource is cluster-scoped, otherwise, the field should not appear in an APIServiceExport object. So I made a few changes in the added commit to improve.

Regarding defaulting of the field, I’m trying to follow. Are you referring to https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting, so that we rely on the schema to set a default value for the field? If that’s the case, we still need somewhere in code (which can be the backend, IMO) to overwrite the default when we want a non-default value, no?

Reply:
Yes, that makes sense.

I think CEL validations for the 1st line is there already. So I added that for the 2nd line in [the last commit](https://github.com/kube-bind/kube-bind/pull/188/commits/17659680e35c35f27d85127693e3e222f1a8f4cb).

If my understanding is still not aligned, please let me know, thank you!